arm-software/embedded/scripts/extract_dmg.sh

```
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Copyright (c) 2025, Arm Limited and affiliates.
+# Part of the Arm Toolchain project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# A bash script for extracting the contents of a MacOS .dmg image.
+
+# The script takes the following arguments:
+# extract_dmg.sh <path_to_dmg> [<glob_pattern_to_extract>] [<dst_dir>]
+# * path_to_dmg: The path to the .dmg image to extract contents from
+# * glob_pattern_to_extract (optional): A glob pattern to match the contents
+#   to be extracted. The default is "*".
+# * dst_dir (optional): The directory into which the contents should be
+#   extracted. The default is the current working dir.
+
+if [[ -z "${1}" ]]; then
+    >&2 echo "ERROR: Path to .dmg file not provided."
+    echo "usage: ${0} <path_to_dmg> [<glob_pattern_to_extract>] [<dst_dir>]"
+    exit 1
+fi
+
+DMG_PATH="${1}"
+PATTERN_TO_EXTRACT="${2:-"*"}"
+DST_DIR="${3:-"."}"
+
+if command -v hdiutil > /dev/null; then
+    ATTACH_OUTPUT=$( hdiutil attach -nobrowse -readonly "${DMG_PATH}" )
+    MOUNT_PATH=$( echo "${ATTACH_OUTPUT}" | grep '/Volumes/' | sed -r 's/.*(\/Volumes\/.*)/\1/' )
+    if [[ -z "${MOUNT_PATH}" ]]; then
+        >&2 echo "ERROR: Path to mounted volume not found."
+        exit 1
+    fi
+    cp -R "${MOUNT_PATH}"/${PATTERN_TO_EXTRACT} .
```

What happens if one of these commands fails? You're not checking the return code of any of them. I _presume_ that if `hdiutil attach` fails then it won't print output matching your regex, so the failure will be detected, but if `cp -R` doesn't work (e.g. because the destination disk has filled up), then the following `hdiutil detach` (assuming it succeeds) will mask the failure by resetting `$?` to 0 before exiting the script.

The usual advice is `set -e` so that the first failing command terminates the script. But in this case I think you probably _do_ still want to do the `hdiutil detach`, because mounting a thing and then forgetting to umount it is also bad error-handling behaviour. I think I'd do something like
```
cp -R "${MOUNT_PATH}"/${PATTERN_TO_EXTRACT} .
exitcode=$?
hdiutil detach "${MOUNT_PATH}" && exit $?
```
which will still _run_ the detach, but then propagate the exit code from the cp.